### DISCUSSION TITLE
Black screen localboot

### DISCUSSION THREAD

**Discussion by ClaudiuLSH** [cobbler/cobbler#2580] [Q&A]
Hi,
I'm facing problems with new hardware and CentOS 7.
The bios is set on legacy mode and the OS installs without problems. If I boot from disk, CentOS loads fine. If I boot from pxe and select local boot, I get a black screen with a blinking cursor.
If I use UEFI the boot order is changed after OS installation finishes and pxe is no longer the first option.

Any ideas on how to make this work?

**Comment by SchoolGuy** [comment]
@ClaudiuPID If my memory does not betray me this is intended behavior. But let me check this if I am back at work, there I have some test machines.

**Comment by ClaudiuLSH** [comment]
UEFI bios update after OS install I understand that it is intended behavior. But pxe boot from local showing black screen with blinking cursor?
I am sure this worked in the past with older hardware.
Or maybe I am doing something wrong..

**Comment by SchoolGuy** [comment]
@ClaudiuPID I will take a look at work asap. But currently I don't have the hardware here to create that scenario.

**Comment by watologo1** [comment]
If I use UEFI the boot order is changed after OS installation finishes and pxe is no longer the first option.
> This certainly is intended. In UEFI the OS can tell the BIOS which boot entry to load next by default and it's probably going to do this. We have a server setup where we have an ugly after_install hack to switch back to network boot, as we need network boot as jump in entry...

Using:
efibootmgr
you can switch boot entries on UEFI in Linux.
You may also want to have a look at the vfat partition BIOS and OS communicate through.
Should be here:
find /boot/efi/

**Comment by watologo1** [comment]
Black screen is certainly not intended behaviour.
Best check /var/log/tftp (or whereever), to follow which boot loader finally is loaded and whether it got executed and tried to load a  config etc....
If it can/could load a config, have a look at it.

**Comment by ClaudiuLSH** [comment]
```
Jul  5 15:48:20 cobbler in.tftpd[6167]: RRQ from 176.124.110.246 filename menu.c32
Jul  5 15:48:20 cobbler in.tftpd[6167]: Client 176.124.110.246 finished menu.c32
Jul  5 15:48:20 cobbler in.tftpd[6176]: RRQ from IP filename pxelinux.cfg/default
Jul  5 15:48:20 cobbler in.tftpd[6176]: Client IP finished pxelinux.cfg/default
```

In `pxelinux.cfg/default` I have for local boot:
```
LABEL local
        MENU LABEL (local)
        MENU DEFAULT
        LOCALBOOT -1
```
I am at a loss here with this setup, due to a bug in a motherboard which enters a pxe boot loop I can't get past this.

**Comment by watologo1** [comment]
Hmm, I never used menu.c32..., this is a separate boot loader?
Better try pxelinux.0
And if this one has issues..., the older the better.
I realized that pxelinux.0 binary doubled size from SLE 10 to SLE 11 to SLE 12...
The older the more robust if you have BIOS issues.
You may want to try an older pxelinux.0 binary you find somewhere in the net, place it into the cobbler deploy directory and do: cobbler sync to get it into the tftp dir (or just try by hand).

You can also try the grub2 binary:
/srv/tftpboot/grub/grub.0
This one is also for legacy i386, but rather untested. Once it loads grub.cfg in there you can add:
set debug=all and you should see something.

Here is my minimal pxelinux.0 part for local boot:
(the network entry chainloads into grub2... not sure it helps, I just paste it to show the syntax).

default local
timeout 10
prompt 1

label local
  localboot 0

label network
  kernel grub2/grub.0

label memtest
  kernel memtest.bin

**Comment by watologo1** [comment]
If you have an old machine or a server with a serial console (ttyS0/ttyS1) you can add:
serial 0 115200

This must be on top of the pxelinux config file and you might loose screen (which you do not have...).

**Comment by SchoolGuy** [comment]
@ClaudiuLSH Were you able to fix this or is this still a problem. If it is a problem: What can Cobbler do for you to improve the situation?
